src/wmtk/simplex/SimplexCollection.hpp

```diff
@@ -0,0 +1,75 @@
+#pragma once
+
+#include <vector>
+#include <wmtk/Mesh.hpp>
+#include "Simplex.hpp"
+#include "internal/SimplexEqualFunctor.hpp"
+#include "internal/SimplexLessFunctor.hpp"
+
+namespace wmtk::simplex {
+class SimplexCollection
+{
+public:
+    SimplexCollection(const Mesh& mesh)
+        : m_mesh{mesh}
+        , m_simplex_is_less(mesh)
+        , m_simplex_is_equal(mesh)
+    {}
+
+    /**
+     * @brief Return const reference to the simplex vector.
+     */
+    const std::vector<Simplex>& simplex_vector() const { return m_simplices; }
```

Comment:
i think this should be `as_vector()`. You definitely don't need to specify that SimplexCollection can return data as a vector.

Reply:
as_vector is the convention from the ranges library, which is a bit more clear imo?